poller/index.js

```diff
@@ -255,22 +255,33 @@ async function parseAndEnrichPayload(payload) {
     }
     else if(spanners[sIdx].units.toUpperCase() == 'NODES') {
       spanners[sIdx].units = spanners[sIdx].units.toUpperCase();
-      // merge in the processing units defaults
-      spanners[sIdx] = {...nodesDefaults, ...spanners[sIdx]};
 
-      // if minNodes and maxNodes are not default and differ from minSize/maxSize
-      // provide a deprecation warning and set minSize/maxSize for the scaler function
-      if(spanners[sIdx].minNodes && spanners[sIdx].minSize != spanners[sIdx].minNodes) {
+      // if minNodes or minSize are provided set the other, and if both are set and not match throw an error
+      if(spanners[sIdx].minNodes && !spanners[sIdx].minSize) {
         log(`DEPRECATION: minNodes is deprecated, remove minNodes from your config and instead use: units = 'NODES' and minSize = ${spanners[sIdx].minNodes}`,
-           'WARNING');
+        'WARNING');
         spanners[sIdx].minSize = spanners[sIdx].minNodes;
+      } else if(spanners[sIdx].minSize && !spanners[sIdx].minNodes) {
+        log(`Setting minNodes to minSize for backward compatibility`, 'INFO');
```

Comment:
This `else if` is not necessary because the scaler does not use minNodes.
I think it is better remove the `else if` to simplify the code
The minNodes/maxNodes properties can then also be removed from nodesDefaults.

Reply:
I think the risk is minimal, but there is the potential for the poller to send data to the scaler that could be missing required fields if both functions have not been upgraded.  The benefits of simplifying that logic outweigh that risk, so I've remove that `else if` and properties from `nodeDefaults`